coursesテーブルの受講期限カラム反映→講師側 講座一覧APIの修正  JKA-1522（もりお）

Morio151515 opened the pull request:
# issue
- Closes  #1045

# 概要
- `courses` テーブルに受講期限カラム（`attendance_deadline`）を追加しました。
- 講師側の講座一覧API（`CourseIndexResource`）に `attendance_deadline` を 返す処理を追加しました。

# 変更内容
- `2022_10_19_215031_create_courses_table.php` の `status` カラムの直後に `attendance_deadline` カラムを追加しました。
- `app/Http/Resources/Instructor/CourseIndexResource.php` に `attendance_deadline` を返す処理を追加しました。
- ご指摘いただきました箇所を修正しました。

# 動作確認
Postman にて`/api/v1/instructor/course/index` を`GET`リクエストで確認し、`attendance_deadline` が `null` と有効な日付の両方で表示されることを確認しました。

# 確認して欲しいこと
コードの記載方法等に誤りがないか等、ご確認いただけましたらと思います。
よろしくお願いいたします。

tatsunori-1550 committed:
coursesテーブルの受講期限カラム反映講師側 講座編集(詳細)APIの修正

Morio151515 committed:
Add attendance_deadline column to courses table

tatsunori-1550 committed:
showメソッドへ修正

Morio151515 committed:
course-deadline-instructor-list

Morio151515 committed:
Merge branch 'develop' into feature/mitsunobu/JKA-1522/course-deadline-instructor-list

yukihiroLaravel commented on app/Http/Resources/Instructor/CourseIndexResource.php:
```diff
@@ -27,6 +27,7 @@ public function toArray($request)
             'status' => $this->resource->status,
             'has_active_students' => (bool) $this->resource->has_active_students,
             'tags' => TagResource::collection($this->resource->tags),
+            'attendance_deadline' => $this->resource->attendance_deadline,
```
日時が存在する場合は、Y-m-d H:i:sというフォーマット変換するようにお願いします！

Morio151515 replied:
`use Carbon\Carbon;`
という `use` を `use` に追加して、
`'attendance_deadline' => $this->resource->attendance_deadline,`
を、
```
'attendance_deadline' => $this->resource->attendance_deadline
    ? Carbon::parse($this->resource->attendance_deadline)->format('Y-m-d H:i:s')
    : null,
```
と書き換えました。
エラーは出ませんでした。
こちらで良いのでしょうか？

Morio151515 replied:
ちゃこさんのコード
 'attendance_deadline' => $this->resource->attendance_deadline
                ? $this->resource->attendance_deadline->format('Y-m-d H:i:s')
                : null,
 に代えてみます！

yukihiroLaravel reviewed (changes requested):
いい感じですが少しコメントしました！
ご確認ください！

Morio151515 committed:
Format attendance_deadline as Y-m-d H:i:s

Morio151515 committed:
Cast attendance_deadline as datetime and format in resource

Morio151515 committed:
Format attendance_deadline as Y-m-d H:i:s

Morio151515 committed:
Format attendance_deadline as Y-m-d H:i:s

github-actions[bot] committed:
Automated code format

Commit:
Add attendance deadline to courses in seeder and update API response structure

- Added 'attendance_deadline' field to course entries in CoursesSeeder.php, setting it to 30 days from now.
- Updated JSON structure assertions in ShowTest.php for both Instructor and Manager to include 'attendance_deadline'.
- Made minor formatting adjustments in route comments for consistency.

Morio151515 committed:
Format attendance_deadline as Y-m-d H:i:s

Morio151515 committed:
Fix: implement deadline check and resolve PHPStan collection error

Morio151515 committed:
Fix: implement deadline check and resolve PHPStan collection error

Morio151515 committed:
Fix: implement deadline check and resolve PHPStan collection error
